src/databricks/labs/ucx/azure/resources.py

```diff
@@ -435,27 +437,52 @@ def _role_assignment(
         scope = assignment_properties.get("scope")
         if not scope:
             return None
-        role_name = self._role_name(role_definition_id)
-        if not role_name:
+        role_details = self._role_name(role_definition_id)
+        if not role_details['roleName']:
             return None
+        role_name = str(role_details['roleName'])
+        role_type = str(role_details['roleType'])
+        role_permissions = role_details['rolePermissions'] if isinstance(role_details['rolePermissions'], list) else []
         principal = self._get_principal(principal_id)
         if not principal:
             return None
         if scope == "/":
             scope = resource_id
         return AzureRoleAssignment(
-            resource=AzureResource(resource_id), scope=AzureResource(scope), principal=principal, role_name=role_name
+            resource=AzureResource(resource_id),
+            scope=AzureResource(scope),
+            principal=principal,
+            role_name=role_name,
+            role_type=role_type,
+            role_permissions=role_permissions,
         )
 
-    def _role_name(self, role_definition_id) -> str | None:
+    def _role_name(self, role_definition_id) -> dict[str, str | list[str] | None]:
         if role_definition_id not in self._role_definitions:
             role_definition = self._mgmt.get(role_definition_id, "2022-04-01")
             definition_properties = role_definition.get("properties", {})
             role_name: str = definition_properties.get("roleName")
             if not role_name:
-                return None
-            self._role_definitions[role_definition_id] = role_name
-        return self._role_definitions.get(role_definition_id)
+                return {
```

Comment:
should make this a dataclass instead of a dict

Reply:
@nkvuong - Done. Please check.